### DISCUSSION TITLE
IPinfo IP-to-country

### DISCUSSION THREAD

**Discussion by junekiwi** [InternetHealthReport/internet-yellow-pages#150] [New dataset] [answered]
Hi IHR team 👋 

What a great project!

Is there any interest in adding IP-to-country geolocation? We (IPinfo) offer a [free dataset](https://ipinfo.io/products/free-ip-database) (CC BY-SA) that is basically `range,country,country_name,continent,continent_name`. This is the same accuracy as our paid offering, it's just the output that is restricted to country-level instead of city/postcode.

AFAICT from [data-sources.md](https://github.com/InternetHealthReport/internet-yellow-pages/blob/main/documentation/data-sources.md) there is currently no geolocation information so I'm not sure how that would fit into the data model but I can play a bit with it to see.

Happy to write the code if there's interest :)

Best,
Max

**Comment by romain-fontugne** [comment]
Thanks Max for starting this discussion. 

Yes, indeed, there is a lot of interest from us to add the IPinfo free dataset (I believe it is on one of our todo lists :) )

For the modeling, I think we can reuse the COUNTRY relationship and Country nodes. The main complication is for the IP ranges. We currently only have prefixes and IP addresses in IYP. So maybe we'd have to translate the IP ranges to prefixes. Do you think this makes sense?

If we could do that then the modeling would be easy:

```
(:Prefix)-[:COUNTRY]-(:Country)
```

**Reply by junekiwi** [reply]
Thanks Romain, that makes sense :) Just to be sure, by prefix do you mean any range in CIDR notation (e.g. `a.b.c.d/x`)? Or is the meaning more specific, like a BGP prefix (in which case one BGP prefix might cover multiple IP locations)?

**Comment by romain-fontugne** [comment]
Yes, I meant a  CIDR notation. Currently most of our prefixes are from BGP but we are in the process of being more explicit by adding different types of Prefixes (e.g. BGP, geolocation, registration). This dataset is a good motivation to implement these changes!

**Comment by m-appel** [comment, marked as answer]
This crawler is now merged into main with 45362e6ec700b4c261383108a35f89791f25e12d.

**Reply by junekiwi** [reply]
Super cool! Thanks for implementing the required features in IYP :)
